### ISSUE TITLE
flax.linen.Conv docs don't mention channel dimension nor dimension numbers generally

### ISSUE THREAD

**Issue by grahamgower** [google/flax#1652]
The [Conv API docs](https://flax.readthedocs.io/en/latest/_autosummary/flax.linen.Conv.html#flax.linen.Conv) don't indicate the convention used for axes ordering. I expected to see something like "we use the channels-last convention (NHWC for a 2d convolution)". Did I miss something obvious?

I eventually figured it out from staring at the flax sources and some jax [conv](https://jax.readthedocs.io/en/latest/jax.lax.html#jax.lax.ConvDimensionNumbers) [docs](https://jax.readthedocs.io/en/latest/_autosummary/jax.lax.conv_general_dilated.html), but this is fairly hard to grok for a newcomer.
https://github.com/google/flax/blob/004a118b8114f8ad9b33bc8fad92ce528a5d022e/flax/core/nn/linear.py#L158-L164

**Comment by EngineerKhan** [comment]
I faced the same issue and took me a long time to figure it out myself (thanks to the flax guy for answering, but it was too late and already took my time). Yeah, it should be part of the docs.

**Comment by jheek** [comment]
I see your point. Actually we do mention the axis ordering in the __call__ docstring but do an issue with the doc generation __call__ so it doesn't show up on ReadTheDocs.

cc @marcvanzee can we fix the doc rendering anytime soon or should we just add the important stuff in the Module docstring as a workaround?

**Comment by marcvanzee** [comment]
Yes i will work on this first thing during our next Flax Docs Tuesday!

**Comment by jheek** [comment]
> Yes i will work on this

And this is fixing the missing `__call__` from RTD?

**Comment by marcvanzee** [comment]
Indeed, I actually have an open PR for this but there were some comments that I should address: #1416

**Comment by grahamgower** [comment]
In addition to having the `__call__` docstring appear in the docs, I think that docstring could probably be a bit more detailed. The dimensions for `flax.linnen.Conv` are different to the defaults used in `jax.lax.conv_general_dilated()`, and this should probably be highlighted too. The docstring is currently quite terse (when looking in the sources for this info, I either didn't grok it enough to realise that was the info I wanted, or I just plain missed it). https://github.com/google/flax/blob/d6a219433ab7a946aa18b416148d7381d65dc5b4/flax/linen/linear.py#L238-L246

**Comment by marcvanzee** [comment]
> In addition to having the `__call__` docstring appear in the docs, I think that docstring could probably be a bit more detailed. The dimensions for `flax.linnen.Conv` are different to the defaults used in `jax.lax.conv_general_dilated()`, and this should probably be highlighted too. The docstring is currently quite terse (when looking in the sources for this info, I either didn't grok it enough to realise that was the info I wanted, or I just plain missed it).

Yes, I agree. I've improved it in #1672. Thanks for the feedback! I'll keep this issue open until my other PR is in.
